addons/l10n_cl/models/account_move.py

```diff
@@ -169,5 +206,104 @@ def _l10n_cl_include_sii(self):
+                    if other_currency:
+                        values['second_currency']['subtotal_amount_taxable'] += line[key_other_currency] * sign_main_currency
+                elif not line.tax_ids:
+                    values['subtotal_amount_exempt'] += line[key_main_currency] * sign_main_currency
+                    if other_currency:
+                        values['second_currency']['subtotal_amount_exempt'] += line[key_other_currency] * sign_main_currency
+        values['global_discounts'] = []
+        for gd in global_discounts:
+            main_value = currency_round_main_currency.round(abs(gd.price_subtotal)) if \
+                (not other_currency and not export) or (other_currency and export) else \
+                currency_round_main_currency.round(abs(gd.balance))
+            second_value = currency_round_other_currency.round(abs(gd.balance)) if other_currency and export else \
+                currency_round_other_currency.round(abs(gd.price_subtotal))
+            values['global_discounts'].append(
+                {
+                    'name': gd.name,
+                    'global_discount_main_value': main_value,
+                    'global_discount_second_value': second_value if second_value != main_value else False,
+                    'tax_ids': gd.tax_ids,
+                }
+            )
+        values['vat_percent'] = '%.2f' % vat_percent if vat_percent > 0 else False
+        return values
+
+    def _l10n_cl_get_withholdings(self):
+        """
+        This method calculates the section of withholding taxes, or 'other' taxes for the Chilean electronic invoices.
+        These taxes are not VAT taxes in general; they are special taxes (for example, alcohol or sugar-added beverages,
+        withholdings for meat processing, fuel, etc.
+        The taxes codes used are included here:
+        [15, 17, 18, 19, 24, 25, 26, 27, 271]
+        http://www.sii.cl/declaraciones_juradas/ddjj_3327_3328/cod_otros_imp_retenc.pdf
+        The need of the tax is not just the amount, but the code of the tax, the percentage amount and the amount
+        :return:
+        """
+        self.ensure_one()
+
+        tax = [{'tax_code': line.tax_line_id.l10n_cl_sii_code,
+                'tax_name': line.tax_line_id.name,
+                'tax_base': abs(sum(self.invoice_line_ids.filtered(
+                    lambda x: line.tax_line_id.l10n_cl_sii_code in x.tax_ids.mapped('l10n_cl_sii_code')).mapped(
+                    'balance'))),
+                'tax_percent': abs(line.tax_line_id.amount),
+                'tax_amount_currency': self.currency_id.round(abs(line.amount_currency)),
+                'tax_amount': self.currency_id.round(abs(line.balance))} for line in self.line_ids.filtered(
+            lambda x: x.tax_group_id.id in [
+                self.env.ref('l10n_cl.tax_group_ila').id, self.env.ref('l10n_cl.tax_group_retenciones').id])]
```

Comment:
there is no way to try to use the tax codes themselves instead of the tax group? (if someone does custom tax groups, nothing will work anymore)

Reply:
You are right. Nevertheless, it was made that way to have some kind of consistency with other l10ns. But perhaps we can do some work around this.